controllers/installation_controller.go

```diff
@@ -229,6 +235,112 @@ func (r *InstallationReconciler) ReconcileK0sVersion(ctx context.Context, in *v1
 	return nil
 }
 
+// ReconcileHelmCharts reconciles the helm charts from the Installation metadata with the clusterconfig object.
+func (r *InstallationReconciler) ReconcileHelmCharts(ctx context.Context, in *v1beta1.Installation) error {
+	var clusterconfig k0sv1beta1.ClusterConfig
+
+	// add k0s scheme to our client
+	k0sv1beta1.AddToScheme(r.Scheme)
+
+	// fetch the current clusterconfig
+	if err := r.Get(ctx, client.ObjectKey{Name: "k0s", Namespace: "kube-system"}, &clusterconfig); err != nil {
+		return fmt.Errorf("failed to get cluster config: %w", err)
+	}
+
+	meta, err := release.MetadataFor(ctx, in.Spec.Config.Version)
+	if err != nil {
+		return fmt.Errorf("failed to get release bundle: %w", err)
+	}
+
+	if meta.Configs == nil {
+		return fmt.Errorf("failed to get helm charts from release metadata")
+	}
+
+	// get the protected values from the release metadata
+	protectedValues := map[string][]string{}
+	if meta.Protected != nil {
+		protectedValues = meta.Protected
+	}
+
+	// TODO - apply unsupported override from installation config
+
+	finalConfigs := k0sv1beta1.ChartsSettings{}
+
+	for _, chart := range clusterconfig.Spec.Extensions.Helm.Charts {
+		for _, newChart := range meta.Configs.Charts {
+			if chart.Name == newChart.Name {
```

Comment:
The code here would be easier to read if the logic was the other way around, I mean:

```go
if chart.Name != newChart.Name {
      continue
}
```